### ISSUE TITLE
"Connection failed, unable to load content" and "User doesn't exist" errors

### ISSUE THREAD

**Issue by georgebvro** [fgl27/smarttv-twitch#298]
Hello,

There is an issue suddenly starting a couple of days ago. I was actually watching a channel and when I exited the live stream and went to the menu, the problems started.

When navigating to User Menu -> Live, User Menu -> Channels, Main Menu -> Live, Main Menu -> Games etc., so pretty much any screen when something must load, an error message is displayed: "Connection failed, unable to load content. Hit refresh to try again".

I deleted my user in the app with the intention of reconnecting it but now when I enter the user it says "User doesn't exist". The user exists though, I tried relogging with it on the Twitch website and it works fine. However, I've created a new user on Twitch website just to try with it also and the same thing happens when I use the "Add User" functionality in the app: "User doesn't exist", so I can't reach the screen where I could enter the authentication key.

I wonder if anyone else has these issues and what else I could try.

Device: Samsung 40JU7000
OS: Tizen 2.3
App: 4.0.1.December 01 2023 (as reported in the app)

Thanks for supporting this project!

**Comment by pnrxtx** [comment]
Yea "connection failed" since some days now.

**Comment by fgl27** [comment]
In fine for me. With and without a user.

Are we OK?

If you don't add a user you run in guess mode, all users in guess mode share the same authentication, witch is limited, if too many are using the app you get those errors.

But that is for who is using without a user.

So let me know if we are OK or not?

**Comment by pnrxtx** [comment]
I'm signed in but "connection failed, unable to load content ...."

**Comment by georgebvro** [comment]
Sorry for the late reply but, unfortunately, it's the same situation.

- When using **Add User** functionality, after entering a valid, existing user, the message "User doesn't exist" is displayed and it doesn't get to the screen where I could enter the authentication key
- **Main Menu -> Live** gives message "Connection failed, unable to load content. Hit refresh to try again" and doesn't display anything
- **Main Menu -> Featured** doesn't give error and displays 10 streams that can be viewed
- **Main Menu -> Games** gives message "Connection failed, unable to load content. Hit refresh to try again" and doesn't display anything
- **Main Menu -> Videos** doesn't give error and displays some VoDs that can be viewed
- **Main Menu -> Clips**  doesn't give error and displays some Clips that can be viewed

Because I can't add a user I can't check User Menu functionality but before I removed the user it didn't work as well (I remember **User Menu -> Live** and **User Menu -> Channels** didn't work for sure).

Let me know how can I help with diagnosing the problem (logs, debugging etc.).

Thank you for the help!

**Comment by fgl27** [comment]
Something on your network is blocking 

For me is fine all of those steps you did all fine.

You can try to debug the app to see if something comes up

Do you know how to debug?

**Comment by georgebvro** [comment]
No, I don't know how. Are there instructions somewhere?

**Comment by fgl27** [comment]
Region on the globe (the flat for some😁 ) ?

so we know if it's a region thing.

**Comment by georgebvro** [comment]
Europe, Romania.

**Comment by fgl27** [comment]
Here is the how to log

https://github.com/fgl27/smarttv-twitch/issues/250#issuecomment-1328306442

Once the app load, in the console tab you must see some red lines

Those usually have a button that can be clicked to expand

That may show the issue

If not I'll add better logging but first I need to set where the logs are coming from

**Comment by georgebvro** [comment]
Ok, I'll follow instructions to debug but I can't do it today, I'll have to try tomorrow.

**Comment by georgebvro** [comment]
When I start the app in Debug mode, I get this error in Tizen Studio.
![image](https://github.com/user-attachments/assets/9398f96b-0816-4ded-9905-2d28adfb8c46)

Starting the app in normal mode works fine (but the same problems described in this issue remain).
I'll try more tomorrow (I have an old version of Tizen Studio installed - v3.2) and old certificates as well. Probably old TV-Extensions too.

Maybe some of these cause the failure to start in Debug mode, even though it starts in normal mode just fine. If you have other ideas about why it might fail, please let me know.

Thanks!

**Comment by NSV-dev** [comment]
Hey same issue here i've got logs

![image](https://github.com/user-attachments/assets/13ea6129-e191-466c-a8b5-47200b3061ae)

**Comment by NSV-dev** [comment]
Here i'm trying to login
![image](https://github.com/user-attachments/assets/005824bf-b462-4f1a-bc02-28435820ba40)

**Comment by NSV-dev** [comment]
And just in case if you need this)
![image](https://github.com/user-attachments/assets/c76134b9-7a63-4481-ae4b-eab85d08ed7f)

**Comment by fgl27** [comment]
I see https issue.

Some Samsung tv have that

There is a settings option **Force http** set it to yes and let me know the result

**Comment by NSV-dev** [comment]
Here is smth interesting "Featured" pages works fine and i even got some random stream
![image](https://github.com/user-attachments/assets/4177da79-432d-4e99-8474-ecb472d1e4d2)

**Comment by NSV-dev** [comment]
> I see https issue.
> 
> Some Samsung tv have that
> 
> There is a settings option **Force http** set it to yes and let me know the result

after switching to yes nothing changed and calls are still https on console

**Comment by fgl27** [comment]
Different screens use different APIs

Does the http workaround works or not?

**Comment by NSV-dev** [comment]
> Different screens use different APIs
> 
> Does the http workaround works or not?

no, i switched it. do i need to restart after that or smth else to save options?

**Comment by fgl27** [comment]
You have to change the setting after installing the app

After each installation the settings reset

**Comment by NSV-dev** [comment]
well in settings it says "yes" but nothing changes

**Comment by NSV-dev** [comment]
same 
![image](https://github.com/user-attachments/assets/a6b25015-7234-4fe2-94e4-299f1dd7db4c)

**Comment by NSV-dev** [comment]
I don't think it's smth on my side. It started few days ago and you can see that at least 2 more people having same issue. I'm from Russia, george is from Romania. Different networks at all, same os and same issue

**Comment by DeneXX1** [comment]
Got same Problem 
Im from Germany

**Comment by georgebvro** [comment]
Somehow it managed to work today in Debug mode and I have identical logs with @NSV-dev.

**Comment by fgl27** [comment]
The issue is on the TV, Samsung is causing this with their silent updates that disable support of some HTTPS calls, they affect more than just this app so eventually they push a fix.

In the meantime, the HTTP workaround was only for the player URLs, not for the general content API.

I enabled it for the falling API, but it isn't a 100% fix as it may not work.

try and let me know.

**Comment by NSV-dev** [comment]
where to find fixed version?

**Comment by fgl27** [comment]
> where to find fixed version?

the app receives when you open it

**Comment by fgl27** [comment]
I have updated the version date in about of the app so you can have a tell

**Comment by georgebvro** [comment]
I've started the app but it only shows a black screen. It has been like this for a few minutes.

**Comment by NSV-dev** [comment]
yep same

**Comment by NSV-dev** [comment]
debug console is empty and screen is just black

**Comment by fgl27** [comment]
is fine for me

**Comment by fgl27** [comment]
try a power off from the wall

**Comment by georgebvro** [comment]
Sadly, that doesn't fix the problem. Screen still black after starting the app and the debug console has not a single message.

**Comment by NSV-dev** [comment]
I disconected on connect box from tv, can't disconect tv without unmounting it, didn't work

**Comment by fgl27** [comment]
I revert the http changes to general API, does it works now?

**Comment by NSV-dev** [comment]
no chnges for me, still black screen

**Comment by fgl27** [comment]
not even a spinner in the center?

**Comment by georgebvro** [comment]
Nope, absolutely nothing.

**Comment by NSV-dev** [comment]
no, but after a while there is a sound like switching between buttons

**Comment by fgl27** [comment]
may is needed to replace all 'https://fgl27.github.io' with 'http://fgl27.github.io'

remove the s

**Comment by NSV-dev** [comment]
yes it helps, changed all github to http and now i have previous problem again

**Comment by fgl27** [comment]
> yes it helps, changed all github to http and now i have previous problem again

can you close the app and try again now?

I push a change I need to know if that bricks the app

**Comment by NSV-dev** [comment]
black screen again

**Comment by fgl27** [comment]
one sec

**Comment by fgl27** [comment]
OK, there was a dependency bugging out the files in git, not related to the problem of the HTTP...

So I fixed the dependency to avoid a black screen


and push the HTTP changes

please test now and let me know the result

**Comment by NSV-dev** [comment]
![image](https://github.com/user-attachments/assets/42541cee-a7a9-4b9d-82d3-b86d4604c566)

**Comment by NSV-dev** [comment]
this is with force http "yes"

**Comment by fgl27** [comment]
> this is with force http "yes"

are you sure

for me shows like this

![image](https://github.com/user-attachments/assets/6b635f6d-d6e9-40cc-81f5-9caabcd0fe0a)

**Comment by NSV-dev** [comment]
maybe I'm doing smth wrong) https://imgur.com/a/ov0DV1v

**Comment by fgl27** [comment]
> maybe I'm doing smth wrong) https://imgur.com/a/ov0DV1v

seems right

what version shows in the about of the app... it must be **4.0.1 September 17 2024 - 3**

seems that you aren't receiving the update

**Comment by NSV-dev** [comment]
there is september 17 in logs

**Comment by NSV-dev** [comment]
yes my version is 4.0.1 September 17 2024 - 3

**Comment by fgl27** [comment]
I don't know way doesn't change then for me changes just fine... let me see if I can add some extra logs one sec

![image](https://github.com/user-attachments/assets/191baf13-5753-47cf-ac4c-5b4a88dd4a92)

**Comment by fgl27** [comment]
I changed to 4.0.1 September 17 2024 - 4 try again see if there is an new log when you switch to http

**Comment by NSV-dev** [comment]
thats funny)
![image](https://github.com/user-attachments/assets/9b24997e-2cce-46f0-b6c8-6df425451d6f)

**Comment by fgl27** [comment]
the SSL error is gone, but the device can't make a request to that API using HTTP only

**Comment by fgl27** [comment]
Not much else to do here

 we need wait Samsung to update the device certificates, which will fix the SSL handshake issue

**Comment by fgl27** [comment]
In the meantime I can migrate some API to working API when available, when I do I'll update here

**Comment by DeneXX1** [comment]
> Not much else to do here
> 
> we need wait Samsung to update the device certificates, which will fix the SSL handshake issue

Maybe it will help you in some way the featured streams in the main menu are working with force http "yes"

Could you perhaps temporarily install a function where you can open the stream directly by typing in the streamer name?

**Comment by fgl27** [comment]
> > Not much else to do here
> > we need wait Samsung to update the device certificates, which will fix the SSL handshake issue
> 
> Maybe it will help you in some way the featured streams in the main menu are working with force http "yes"
> 
> Could you perhaps temporarily install a function where you can open the stream directly by typing in the streamer name?
> 
> So the player opens directly when you type in the name of the live stream, maybe this work?

I don't wanna to add workarounds as the one you request takes more time to me to make then to migrate a API for example to the search API that allows to search, Lives, videos, channels or games

**Comment by fgl27** [comment]
I don't have much time that is why I have not updated this...

if anyone has just updated the app, this is an open-source project and the Android version of the app https://github.com/fgl27/SmartTwitchTV has all the necessary updates in place.

**Comment by DeneXX1** [comment]
Is it working again for anyone?
The app hasn't worked for me since then and Samsung won't release any updates for the old Tvs either im Sure

**Comment by pnrxtx** [comment]
I still launch app every 3 days hoping for a fix that it starts working. It was perfect for 2015 year tv, couldn't be better when it was working.

**Comment by fgl27** [comment]
> Is it working again for anyone? The app hasn't worked for me since then and Samsung won't release any updates for the old Tvs either im Sure

the update if they do will be silent.

anyway my above post continues, anyone who has time can work on migrating API requests, I don't

**Comment by noriea** [comment]
Sadly, this started happening for me today. Was working excellent for months, even blocking ads. Was watching streams, turned off tv and came back an hour later and got all the same errors above. Seems like no one took on updating the app, shame. Thanks for what you've done with this app fgl27. It's been great.

**Comment by Teleplox** [comment]
Same here today. I get always "Connection failed, unable to load content" execpt in "Main Menu -> Featured".

**Comment by Teleplox** [comment]
After debugging its this error:

![Image](https://github.com/user-attachments/assets/3a8efffd-2656-41d2-9bc1-3d64703a3444)

I think the TV dont trust the certificate.
Twitch renew the certificate today:

![Image](https://github.com/user-attachments/assets/303616d7-9f4c-49ed-ab81-8e2ce900af15)

I dont know why this should be a problem because the official tizen twitch app works fine and this app will use the same api.

**Comment by fgl27** [comment]
> After debugging its this error:
> 
> ![Image](https://github.com/user-attachments/assets/3a8efffd-2656-41d2-9bc1-3d64703a3444)
> 
> I think the TV dont trust the certificate. Twitch renew the certificate today:
> 
> ![Image](https://github.com/user-attachments/assets/303616d7-9f4c-49ed-ab81-8e2ce900af15)
> 
> I dont know why this should be a problem because the official tizen twitch app works fine and this app will use the same api.

Twitch doesn't use the third party API, and those are the one falling. The feature section uses undocumented API the same the official app uses.

Maybe because this is app calling API from a different domains something in the device checks the app domain with is local vs the API and declare it unsecured.

Maybe there is some in app config that can be changed to fix that. 

Something that goes here for example

https://github.com/fgl27/smarttv-twitch/blob/master/app/config.xml

Doc 

https://developer.samsung.com/smarttv/develop/guides/fundamentals/configuring-tv-applications.html

I don't have any defected devices can't test but if anyone what's to discuss they can, if you wanna to open a separate discussion issue feel free to do so.

**Comment by moooises** [comment]
I had this problem during the weekend, but now, Monday in the morning where I live, I works now.
Maybe the Samsung update the certificates. I don't know the TV model at the moment. When I have access to it, I will report it.

**Comment by noriea** [comment]
Randomly working again today.

**Comment by moooises** [comment]
> I had this problem during the weekend, but now, Monday in the morning where I live, I works now. Maybe the Samsung update the certificates. I don't know the TV model at the moment. When I have access to it, I will report it.

My TV model is Samsung 49" Q70R 4K UHD QLED Smart TV QE49Q70RAT (2019).

**Comment by dfuentes87** [comment]
Mine also started working today.
